cloudferrylib/os/image/glance_image.py

```diff
@@ -373,24 +386,29 @@ def make_image_info(self, glance_image, info):
 
         return info
 
-    def deploy(self, info):
-        LOG.info("Glance images deployment started...")
-        info = copy.deepcopy(info)
-        new_info = {'images': {}}
-        created_images = []
-        delete_container_format, delete_disk_format = [], []
-        empty_image_list = {}
-        keystone = self.cloud.resources["identity"]
-
-        # List for obsolete/broken images IDs, that will not be migrated
-        obsolete_images_ids_list = []
+    def _dst_images(self):
         dst_images = {}
+        keystone = self.cloud.resources["identity"]
         for dst_image in self.get_image_list():
             tenant_name = keystone.try_get_tenant_name_by_id(
                 dst_image.owner, default=self.cloud.cloud_config.cloud.tenant)
             image_key = (dst_image.name, tenant_name, dst_image.checksum,
                          dst_image.is_public)
             dst_images[image_key] = dst_image
+        return dst_images
+
+    def deploy(self, *args):
+        LOG.info("Glance images deployment started...")
+        if len(args) != 1:
+            raise TypeError("deploy() takes exactly 1 argument.")
```

Comment:
In fact it doesn't, instead it only hides the issue. The resolution for that would be to refactor all inherited `deploy()` methods to follow the same signature. Again, type checks must be done by interpreter, not the business logic.